src/io/linux.zig

```diff
@@ -960,6 +960,84 @@ pub const IO = struct {
         self.enqueue(completion);
     }
 
+    pub const INVALID_EVENT = @bitCast(u32, @as(i32, -1));
+
+    pub const EventResponse = enum {
+        listen,
+        cancel,
+    };
+
+    pub fn open_event(
+        self: *IO,
+        completion: *Completion,
+        comptime on_event: fn (*Completion) EventResponse,
+    ) !u32 {
+        // eventfd initialized with no (zero) previous write value,
+        const event_fd = os.eventfd(0, linux.EFD.CLOEXEC) catch |err| switch (err) {
+            error.SystemResources, error.SystemFdQuotaExceeded, error.ProcessFdQuotaExceeded => return error.SystemResources,
+            error.Unexpected => return error.Unexpected,
+        };
+        assert(event_fd != INVALID_EVENT);
+        errdefer os.close(event_fd);
+
+        // Start listening for notifications on this event.
+        const event = @intCast(u32, event_fd);
```

Comment:
Why cast this from `fd_t`?

Reply:
Wanted a unified "event" type across impls so went with `u32`. I can go with i32 so fd_t is compatible.

Reply:
Yea, although I've changed it to be i32 now for compatibility.